crates/bevy_asset/src/server/info.rs

```diff
@@ -562,44 +562,49 @@ impl AssetInfos {
-                if entry.get_mut().handle_drops_to_skip > 0 {
-                    entry.get_mut().handle_drops_to_skip -= 1;
-                    false
-                } else {
-                    let info = entry.remove();
-                    if let Some(path) = info.path {
-                        if watching_for_changes {
-                            for loader_dependency in info.loader_dependencies.keys() {
-                                if let Some(dependants) =
-                                    loader_dependants.get_mut(loader_dependency)
-                                {
-                                    dependants.remove(&path);
-                                }
-                            }
-                            if let Some(label) = path.label() {
-                                let mut without_label = path.to_owned();
-                                without_label.remove_label();
-                                if let Entry::Occupied(mut entry) =
-                                    living_labeled_assets.entry(without_label)
-                                {
-                                    entry.get_mut().remove(label);
-                                    if entry.get().is_empty() {
-                                        entry.remove();
-                                    }
-                                };
-                            }
-                        }
-                        path_to_id.remove(&path);
-                    }
-                    true
-                }
-            }
+        let Entry::Occupied(mut entry) = infos.entry(id) else {
             // Either the asset was already dropped, it doesn't exist, or it isn't managed by the asset server
             // None of these cases should result in a removal from the Assets collection
-            Entry::Vacant(_) => false,
+            return false;
+        };
+
+        if entry.get_mut().handle_drops_to_skip > 0 {
+            entry.get_mut().handle_drops_to_skip -= 1;
+            return false;
+        }
+
+        let info = entry.remove();
+        let Some(path) = info.path else {
+            return true;
+        };
+
+        if !watching_for_changes {
+            path_to_id.remove(&path);
+            return true;
+        }
+
+        for loader_dependency in info.loader_dependencies.keys() {
+            if let Some(dependants) = loader_dependants.get_mut(loader_dependency) {
+                dependants.remove(&path);
+            }
         }
+
+        let Some(label) = path.label() else {
+            path_to_id.remove(&path);
+            return true;
+        };
+
+        let mut without_label = path.to_owned();
+        without_label.remove_label();
+
+        if let Entry::Occupied(mut entry) = living_labeled_assets.entry(without_label) {
+            entry.get_mut().remove(label);
+            if entry.get().is_empty() {
+                entry.remove();
+            }
+        };
```

Comment:
This passage is too difficult to understand and duplicates the `path_to_id.remove(&path)` line, making it more obscure. It is more appropriate to extract it instead of inverting it. It is essentially a sort of subroutine anyway.

Basically you should revert the old `if watching_for_changes` block, and extract all the code inside it to a new function. Then you can invert the body of the new function as you usually do. If you use rust-analyzer there is a specific action to extract selected code into a function.

Once that's done, I think it's probably OK to go.